### PR TITLE
VMware vSAN license module

### PR DESCRIPTION
**Pull request by kamsz**
This PR adds ability to add or remove vSAN license for particular cluster.

Example playbook:

```
- name: Assign license to Management cluster
      local_action:
        module: vmware_vsan_license
        hostname: "{{ vcenter_ip_address }}"
        username: "{{ vcenter_username }}"
        password: "{{ vcenter_password }}"
        datacenter_name: "{{ datacenter_name }}"
        cluster_name: Management
        license_key: "{{ vsan_license_key }}"
        state: present
```

Example run:

```
Using /etc/ansible/ansible.cfg as config file
 [WARNING]: provided hosts list is empty, only localhost is available

Loaded callback default of type stdout, v2.0
1 plays in vmware_vsan_license.yml

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ESTABLISH LOCAL CONNECTION FOR USER: kamil
127.0.0.1 EXEC /bin/sh -c '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1454578980.32-33048092858639 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1454578980.32-33048092858639 )" )'
127.0.0.1 PUT /tmp/tmpGysWbS TO /home/kamil/.ansible/tmp/ansible-tmp-1454578980.32-33048092858639/setup
127.0.0.1 EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/kamil/.ansible/tmp/ansible-tmp-1454578980.32-33048092858639/setup; rm -rf "/home/kamil/.ansible/tmp/ansible-tmp-1454578980.32-33048092858639/" > /dev/null 2>&1'
ok: [localhost]

TASK [Add vSAN license to Management cluster] **********************************
task path: /home/kamil/Documents/repositories/vmware_vsan_license.yml:4
ESTABLISH LOCAL CONNECTION FOR USER: kamil
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1454578980.6-160180590163862 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1454578980.6-160180590163862 )" )'
localhost PUT /tmp/tmpPUfiJx TO /home/kamil/.ansible/tmp/ansible-tmp-1454578980.6-160180590163862/vmware_vsan_license
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/kamil/.ansible/tmp/ansible-tmp-1454578980.6-160180590163862/vmware_vsan_license; rm -rf "/home/kamil/.ansible/tmp/ansible-tmp-1454578980.6-160180590163862/" > /dev/null 2>&1'
changed: [localhost -> localhost] => {"changed": true, "invocation": {"module_args": {"_ansible_check_mode": false, "_ansible_debug": false, "_ansible_diff": false, "_ansible_no_log": false, "_ansible_verbosity": 4, "cluster_name": "Management", "datacenter_name": "Test", "hostname": "10.91.97.30", "license_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "state": "present", "username": "administrator@vsphere.local"}, "module_name": "vmware_vsan_license"}}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @kamsz for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

**Comment by gregdek** [comment]
Thanks @kamsz for this PR. A maintainer of this module has asked for revisions to this PR. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by gregdek** [comment]
@kamsz A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by kamsz** [comment]
ready_for_review :+1:

**Comment by gregdek** [comment]
Thanks @kamsz. To the current maintainers, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/vmware/vmware_vsan_license.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            - Name of the cluster that will be licensed
+    license_key:
+        description:
+            - License key that will be used for vSAN licensing
+        required: True
+    state:
+        description:
+            - If the datacenter should be present or absent
+        choices: ['present', 'absent']
+        required: True
```
Comment (resmo) [inline cloud/vmware/vmware_vsan_license.py]: in many modules, state is defaulted to 'present'. It has become a convention. If you follow this convention remove the `required: true` and add `default: present`

File: cloud/vmware/vmware_vsan_license.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+def main():
+
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        dict(
+                datacenter_name=dict(required=True, type='str'),
+                cluster_name=dict(required=True, type='str'),
+                license_key=dict(required=True, type='str', no_log=True),
+                state=dict(required=True, choices=['present', 'absent'], type='str'),
```
Comment (resmo) [inline cloud/vmware/vmware_vsan_license.py]: as written before, remove required=True and add default=present

**Comment by kamsz** [comment]
State defaults to present now.

**Comment by gregdek** [comment]
Thanks @kamsz for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kamsz A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kamsz Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kamsz A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kamsz Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kamsz A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@kamsz Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Closing due to contributor inactivity.

**Pull request closed without merging**